File: api/spring-boot-api/src/main/java/com/spicytodo/controller/HealthController.java
package com.spicytodo.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;
import java.util.Map;

@RestController
@CrossOrigin(origins = {"http://localhost:3000", "http://127.0.0.1:3000"})
public class HealthController {

    @GetMapping("/")
    public ResponseEntity<Map<String, String>> root() {
        return ResponseEntity.ok(Map.of(
                "message", "🌶️ Spicy Todo API - Spring Boot Implementation",
                "version", "1.0.0",
                "docs", "/api/todos"
        ));
    }

    @GetMapping("/health")
    public ResponseEntity<Map<String, String>> health() {
        return ResponseEntity.ok(Map.of(
                "status", "healthy",
                "service", "spicy-todo-spring-api",
                "uptime", LocalDateTime.now().toString()
        ));
    }
}

File: api/dotnet-api/Controllers/HealthController.cs
using Microsoft.AspNetCore.Mvc;

namespace DotnetApi.Controllers
{
    [ApiController]
    public class HealthController : ControllerBase
    {
        [HttpGet("/")]
        public ActionResult GetRoot()
        {
            return Ok(new
            {
                message = "🌶️ Spicy Todo API - .NET Core Implementation",
                version = "1.0.0",
                docs = "/swagger"
            });
        }

        [HttpGet("/health")]
        public ActionResult GetHealth()
        {
            return Ok(new
            {
                status = "healthy",
                service = "spicy-todo-dotnet-api",
                uptime = DateTime.UtcNow.ToString("O")
            });
        }
    }
}

